micro_sam/evaluation/benchmark_datasets.py

```diff
@@ -0,0 +1,448 @@
+            path=os.path.join(path, "nuc_mm"), sample="zebrafish", download=True,
+        ),
+        "uro_cell": lambda: datasets.uro_cell.get_uro_cell_data(
+            path=os.path.join(path, "uro_cell"), download=True,
+        ),
+        "sponge_em": lambda: datasets.sponge_em.get_sponge_em_data(
+            path=os.path.join(path, "sponge_em"), download=True,
+        ),
+        "vnc": lambda: datasets.vnc.get_vnc_data(
+            path=os.path.join(path, "vnc"), download=True,
+        ),
+        "asem_mito": lambda: datasets.asem.get_asem_data(
+            path=os.path.join(path, "asem"), volume_ids=datasets.asem.ORGANELLES["mito"], download=True,
+        )
+    }
+
+    if dataset_choice is None:
+        dataset_choice = available_datasets.keys()
+    else:
+        if not isinstance(dataset_choice, list):
+            dataset_choice = [dataset_choice]
+
+    for choice in dataset_choice:
+        if choice in available_datasets:
+            available_datasets[choice]()
+        else:
+            raise ValueError(f"'{choice}' is not a supported choice of dataset.")
+
+    return dataset_choice
+
+
+def _extract_slices_from_dataset(path, dataset_choice):
+    if dataset_choice in LM_2D_DATASETS or EM_2D_DATASETS:
+        ndim, tile_shape = 2, (512, 512)
+    else:
+        ndim, tile_shape = 3, (32, 512, 512)
+
+    available_datasets = {
+        # Light Microscopy datasets
+        "livecell": lambda: datasets.livecell._get_livecell_paths(path=path, split="test"),
+        "deepbacs": lambda: datasets.deepbacs._get_deepbacs_paths(path=path, split="test", bac_type="mixed"),
+    }
+
+    image_paths, gt_paths = available_datasets[dataset_choice]()
+    if dataset_choice in DATASET_RETURNS_FOLDER:
+        image_paths = glob(os.path.join(image_paths, DATASET_RETURNS_FOLDER[dataset_choice]))
+        gt_paths = glob(os.path.join(gt_paths, DATASET_RETURNS_FOLDER[dataset_choice]))
+
+    image_paths, gt_paths = natsorted(image_paths), natsorted(gt_paths)
+    assert len(image_paths) == len(gt_paths)
+
+    # Directory where we store the extracted ROIs.
+    save_image_dir = os.path.join(path, f"roi_{ndim}d", "inputs")
+    save_gt_dir = os.path.join(path, f"roi_{ndim}d", "labels")
+    if os.path.exists(save_image_dir) and os.path.exists(save_gt_dir):
+        return save_image_dir, save_gt_dir, ndim
+
+    os.makedirs(save_image_dir, exist_ok=True)
+    os.makedirs(save_gt_dir, exist_ok=True)
+
+    # Logic to extract relevant patches for inference
+    image_counter = 1
+    for image_path, gt_path in tqdm(
+        zip(image_paths, gt_paths), total=len(image_paths),
+        desc=f"Extracting patches for {dataset_choice}"
+    ):
+        image = imageio.imread(image_path)
+        gt = imageio.imread(gt_path)
+
+        if len(np.unique(gt)) == 1:  # There could be labels which does not have any annotated foreground.
+            continue
+
+        tiling = blocking([0, 0], gt.shape, tile_shape)
```

Comment:
This will not work for 3d data, we need to passs `[0, 0, 0]` for this. It's probably best to define a variable `origin` for this. See above.